src/_guides/libraries/library-tour_migrated.md

```diff
@@ -0,0 +1,1707 @@
+
+// Replace every match with another string.
+var exedOut = someDigits.replaceAll(numbers, 'XX');
+assert(exedOut == 'llamas live XX to XX years');
+```
+
+You can work directly with the RegExp class, too. The Match class
+provides access to a regular expression match.
+
+<?code-excerpt "../null_safety_examples/misc/test/library_tour/core_test.dart (match)"?>
+```dart
+var numbers = RegExp(r'\d+');
+var someDigits = 'llamas live 15 to 20 years';
+
+// Check whether the reg exp has a match in a string.
+assert(numbers.hasMatch(someDigits));
+
+// Loop through all matches.
+for (var match in numbers.allMatches(someDigits)) {
+  print(match.group(0)); // 15, then 20
+}
+```
+
+#### More information
+
+Refer to the [String API reference][String] for a full list of
+methods. Also see the API reference for [StringBuffer,][StringBuffer]
+[Pattern,][Pattern] [RegExp,][RegExp] and [Match.][Match]
+
+### Collections
+
+Dart ships with a core collections API, which includes classes for
+lists, sets, and maps.
+
+#### Lists
+
+As the language tour shows, you can use literals to create and
+initialize [lists](#lists). Alternatively, use one of the List
+constructors. The List class also defines several methods for adding
+items to and removing items from lists.
+
+<?code-excerpt "../null_safety_examples/misc/test/library_tour/core_test.dart (List)"?>
+```dart
+// Create a list using a list literal.
+var fruits = ['apples', 'oranges'];
+
+// Add to a list.
+fruits.add('kiwis');
+
+// Add multiple items to a list.
+fruits.addAll(['grapes', 'bananas']);
+
+// Get the list length.
+assert(fruits.length == 5);
+
+// Remove a single item.
+var appleIndex = fruits.indexOf('apples');
+fruits.removeAt(appleIndex);
+assert(fruits.length == 4);
+
+// Remove all elements from a list.
+fruits.clear();
+assert(fruits.isEmpty);
+
+// You can also create a List using one of the constructors.
+var vegetables = List.filled(99, 'broccoli');
+```
+
+Use `indexOf()` to find the index of an object in a list:
+
+<?code-excerpt "../null_safety_examples/misc/test/library_tour/core_test.dart (indexOf)"?>
+```dart
+var fruits = ['apples', 'oranges'];
+
+// Access a list item by index.
+assert(fruits[0] == 'apples');
+
+// Find an item in a list.
+assert(fruits.indexOf('apples') == 0);
+```
+
+Sort a list using the `sort()` method. You can provide a sorting
+function that compares two objects. This sorting function must return \<
+0 for *smaller*, 0 for the *same*, and \> 0 for *bigger*. The following
+example uses `compareTo()`, which is defined by
+[Comparable][] and implemented by String.
+
+<?code-excerpt "../null_safety_examples/misc/test/library_tour/core_test.dart (compareTo)"?>
+```dart
+var fruits = ['bananas', 'apples', 'oranges'];
+
+// Sort a list.
+fruits.sort((a, b) => a.compareTo(b));
+assert(fruits[0] == 'apples');
+```
+
+Lists are parameterized types ([generics]({{site.www}}/guides/language/language-tour#generics)), so you can specify the type that a list
```

Comment:
Fixing the line length and the link. (We no longer use site.www and should remove/replace it. I've filed #3003 to track this.)

```suggestion
Lists are parameterized types
([generics](/guides/language/language-tour#generics)),
so you can specify the type that a list
```